src/pygeosx/pysolver/PySolver.cpp

```diff
@@ -0,0 +1,226 @@
+    else
+    {
+      subEvent = nullptr;
+    }
+  }
+  PYTHON_ERROR_IF( subEvent == nullptr, PyExc_RuntimeError, "Target not found", -1 );
+
+  group = static_cast< geosx::SolverBase * >(subEvent->getEventTarget());
+  self->group = group;
+
+  return 0;
+}
+
+
+static PyObject * PySolver_repr( PyObject * const obj ) noexcept
+{
+  PySolver const * const pySolver = LvArray::python::convert< PySolver >( obj, getPySolverType() );
+  if( pySolver == nullptr )
+  {
+    return nullptr;
+  }
+
+  VERIFY_INITIALIZED( pySolver );
+
+  string const path = pySolver->group->getPath();
+  string const type = LvArray::system::demangle( typeid( *(pySolver->group) ).name() );
+  string const repr = path + " ( " + type + " )";
+
+  return PyUnicode_FromString( repr.c_str() );
+}
+
+
+
+static PyObject * solverStep( PySolver * self, PyObject * args )
+{
+  VERIFY_NON_NULL_SELF( self );
+  VERIFY_INITIALIZED( self );
+
+  double time;
+  double dt;
+  if( !PyArg_ParseTuple( args, "dd", &time, &dt ) )
+  {
+    return nullptr;
+  }
+
+  geosx::DomainPartition & domain = g_state->getProblemManager().getDomainPartition();
+
+  self->group->solverStep( time, dt, 0, domain );
+
+  Py_RETURN_NONE;
+}
+
+
+static PyObject * reinit( PySolver * self, PyObject *args )
+{
+  VERIFY_NON_NULL_SELF( self );
+  VERIFY_INITIALIZED( self );
+  GEOSX_UNUSED_VAR( args );
+
+  self->group->reinit();
+
+  Py_RETURN_NONE;
+}
+
+
+static PyMethodDef PySolver_methods[] = {
+  { "solverStep", (PyCFunction) solverStep, METH_VARARGS, "solver Step" },
+  { "reinit", (PyCFunction) reinit, METH_NOARGS, "re-initialize certain variable depending on the solver being used"},
+  { "get_wrapper", (PyCFunction) PyGroup_getWrapper< PySolver >, METH_VARARGS, PyGroup_getWrapperDocString },
+  { nullptr, nullptr, 0, nullptr }      /* Sentinel */
+};
+
+
+/**
+ * Initialize the module object for Python with the exported functions
+ */
+
+BEGIN_ALLOW_DESIGNATED_INITIALIZERS
+
+static PyTypeObject PySolverType = {
+  PyVarObject_HEAD_INIT( nullptr, 0 )
+    .tp_name = "pysolver.Solver",
+  .tp_basicsize = sizeof( PySolver ),
+  .tp_itemsize = 0,
+  .tp_dealloc = (destructor)PySolver_dealloc,
+  .tp_repr = PySolver_repr,
+  .tp_flags = Py_TPFLAGS_DEFAULT | Py_TPFLAGS_BASETYPE,
+  .tp_doc = PySolver::docString,
+  .tp_methods = PySolver_methods,
+  .tp_init = (initproc)PySolver_init,
+  .tp_new = PySolver_new,
+};
+
+END_ALLOW_DESIGNATED_INITIALIZERS
+
+static PyModuleDef pysolvermodule = {
+  PyModuleDef_HEAD_INIT,
+  "pysolver",
+  "pysolver module for SolverBase",
+  -1,
+  NULL, NULL, NULL, NULL, NULL
+};
+
+
+PyMODINIT_FUNC
+PyInit_pysolver( void )
+{
+  //PyObject * module = PyModule_Create( &pysolvermodule );
+  LvArray::python::PyObjectRef<> module{ PyModule_Create( &pysolvermodule ) };
+  if( module == nullptr )
+  {
+    return nullptr;
+  }
+
+  if( !LvArray::python::addTypeToModule( module, geosx::python::getPySolverType(), "Solver" ) )
+  {
+    return nullptr;
+  }
+
+  return module;
```

Comment:
I'm still not sure why I can't see the time history file. But if you look at how I do it for the LvArray module it should be pretty similar.

https://github.com/GEOSX/LvArray/blob/3ad0795f10023bd9a2fd8a41f0de0caa376845c0/src/python/python.cpp#L162

Specifically you'll see that you return `module.release()`
https://github.com/GEOSX/LvArray/blob/3ad0795f10023bd9a2fd8a41f0de0caa376845c0/src/python/python.cpp#L134

Reply:
You removed `pyhdf5` in 2b6c10cfbe487f46053bc06d1152a063cf59f01f, I imagine you just forgot to add the renamed file.

Reply:
`return module.release()` instead of `return module`, that's exactly what I was missing. It's working now, thanks !

Reply:
Also you're right I forgot to add the new pyhistory directory in the commit... Sorry for that.